website/docs/recipes/kafka.md

```diff
@@ -4,4 +4,168 @@ title: Kafka
+
+    assertDoesNotThrow(() -> {
+        expectApplicationToConsumeKafkaBytesSuccessfully(kafkaBytes);
+    });
+}
+
+private byte[] convertToKafkaBytes(Message message) {
+    return message.contentsAsBytes();
+}
+
+private void expectApplicationToConsumeKafkaBytesSuccessfully(byte[] kafkaBytes) {
+    ConsumerDomainRecord consumerDomainRecord = useProductionCodeToDeserializeKafkaBytesToDomain(kafkaBytes);
+
+    ProductionCode productionCode = new ProductionCode();
+    productionCode.handle(consumerDomainRecord);
+}
+
+private ConsumerDomainRecord useProductionCodeToDeserializeKafkaBytesToDomain(byte[] kafkaBytes) {
+    Deserializer<ConsumerDomainRecord> deserializer = getProductionKafkaDeserializer();
+    return deserializer.deserialize("", kafkaBytes);
+}
+
+private Deserializer<ConsumerDomainRecord> getProductionKafkaDeserializer() {
+    KafkaJsonDeserializer<ConsumerDomainRecord> domainRecordKafkaJsonDeserializer = new KafkaJsonDeserializer<>();
+    Map<String, Object> props = Map.of(
+            KafkaJsonDeserializerConfig.JSON_VALUE_TYPE, ConsumerDomainRecord.class.getName()
+    );
+    domainRecordKafkaJsonDeserializer.configure(props, false);
+    return domainRecordKafkaJsonDeserializer;
+}
+```
+
+Our test method gets a List of Messages passed in - these are PACT's representation of the message(s) that we defined in the previous step. Each message has space for a byte array and also metadata. We're not going to look much at metadata at this point. 
+
+The "useProductionCodeToDeserializeKafkaBytesToDomain" method tries to indicate that we're meant to be using some real production code in our tests. After all, there would be little point if our tests just test themselves. With Kafka, however, this is quite tricky. We need a production object that will take our byte array and process it. With Kafka, we're not so close to the low level action - we typically use "out of the box" classes to do this work for us. A KafkaConsumer will typically return ConsumerRecords<Key,Value> and you can't pass it a byte array! However, it would be configured with the appropriate Deserializers. For normal JSON, we'd configure it with a KafkaJsonDeserializer. Now, those take byte arrays as input. So, here I would suggest that you read your app's configuration, and work our what the class of the deserialiser is, instantiating an instance and returning it. A little tricky, but that's about as close to production code as we're going to get. The "useProductionCodeToDeserializeKafkaBytesToDomain" is a quicker way of doing it for the sake of this demo.
+
+Having turned out byte array into a custom domain object, what do we do? Well, what would your production application do with it? Call that code and make sure it can process it successfully. We don't need to do all the assertions that "service" tests might be interested in, we just need to be sure that our application understood the input. A simple "assertDoesNotThrow" is likely sufficient.
```

Comment:
More code gates?

```suggestion
Having turned out byte array into a custom domain object, what do we do? Well, what would your production application do with it? Call that code and make sure it can process it successfully. We don't need to do all the assertions that "service" tests might be interested in, we just need to be sure that our application understood the input. A simple `assertDoesNotThrow` is likely sufficient.
```

Reply:
Done! :)